tests/test_location_pi.py

```diff
@@ -0,0 +1,665 @@
+        self.create_quant(self.strawberry.id, self.test_location_01.id, 1,
+                          lot_id=lot.id)
+        adjs_req = [{"product_id": self.strawberry.id,
+                     "package_name": "test20_NO_PACKAGE_with_existent_lot",
+                     "quantity": 98,
+                     "lot_name": lot_name}]
+
+        stock_drift = self.test_location_01._get_stock_drift(adjs_req)
+
+        self.assertEqual(len(stock_drift), 1, "Not a unique stock drift entry")
+
+        (stock_info, qty), = stock_drift.items()
+
+        self.assertEqual(qty, 98, "Wrong adjustment quantity")
+        self.assertEqual(stock_info.product_id, self.strawberry.id,
+                         "Wrong product id")
+        self.assertFalse(stock_info.package_id, "Wrongly added a product id")
+        self.assertEqual(stock_info.lot_id, lot.id, "Wrong lot specified")
+
+    def test21_process_inventory_adjustment_mss(self):
+        """
+        Returns the expected stock.inventory.line representing
+        the requested inventory adjustment, after a valid request.
+        """
+        Inventory = self.env['stock.inventory']
+
+        self.create_quant(self.apple.id, self.test_location_01.id, 4)
+        adjs_req = [{"product_id": self.apple.id,
+                     "package_name": "test21_NO_PACKAGE",
+                     "quantity": 3}]
+        expected_name = 'PI inventory adjustment ' + self.test_location_01.name
+
+        inv_adj = self.test_location_01\
+                      ._process_inventory_adjustments(adjs_req)
+
+        self.assertEqual(inv_adj.name, expected_name,
+                         "Unexpected inventory adjustment name")
+
+        inv = Inventory.search([('name', '=', inv_adj.name)])
+
+        self.assertEqual(len(inv), 1, "Did not create the inventory")
+
+        inv_line = inv.line_ids
+
+        self.assertEqual(len(inv_line), 1,
+                         "Did not add a single inventory line")
+        self.assertEqual(inv_line.product_qty, 3, "Wrong quantity")
+        self.assertFalse(inv_line.package_id,
+                         "A package has been wrongly associated")
+        self.assertFalse(inv_line.prod_lot_id,
+                         "A lot has been wrongly associated")
+        self.assertEqual(inv_line.location_id.id, self.test_location_01.id,
+                         "Inventory line associated to the wrong location")
+
+    def test22_process_single_preceding_adjustments_request_mss(self):
+        """
+        Correctly processes the specified preceding adjustment
+        request and links it to the specified inventory adjustment.
+        """
+        Inventory = self.env['stock.inventory']
+
+        package_name = "test22_NO_PACKAGE_with_lot"
+        adjs_req = [{"product_id": self.banana.id,
+                     "package_name": package_name,
+                     "quantity": 28,
+                     "lot_name": "beautiful_lot"}]
```

Comment:
I think banana is not tracked by serial.

Reply:
Indeed, this triggers the `product.tracking` check.